### PR TITLE
Display quota in sidebar and move information about current folder below files list

### PR DESCRIPTION
**Pull request by LukasHirt**
## Related Issue

- Fixes https://github.com/owncloud/product/issues/120
- Fixes https://github.com/owncloud/product/issues/121

## How Has This Been Tested?

- Set quota to user
- Fill quota above 80%
- Fill full files list with resources

## Screenshots (if appropriate):
image
image

### CODE REVIEW

**Comment by pmaier1** [comment]
Great work, love it :+1: I'd remove the "occupied" text as I think that's quite clear and would make it cleaner.

**Comment by pmaier1** [comment]
> Please indicate warning not only with color change. Maybe add additional warning sign…

Even if you can't distinguish the colors you will notice that the bar is quite full, won't you? In addition the text below will tell you.
I really don't think we need another hint for that. Please, keep it simple and clean.

**Comment by michaelstingl** [comment]
@pmaier1 then not use color change at all to treat everyone equal?

With the color change, you want to get extra-attention, so users can have a closer look about the problem. Why exclude people there?

**Comment by pmaier1** [comment]
> @pmaier1 then not use color change at all to treat everyone equal?

If you think that it is more important to "treat everyone equally", then I'd remove the color indication instead of adding another indicator :man_shrugging:

**Comment by pmaier1** [comment]
Ok, so let's not use the color for now to keep things simple without excluding anybody. Maybe later on another cool solution will come up.

**Comment by LukasHirt** [comment]
@pmaier1 @michaelstingl New version without the variation and `occupied` text:

image

**Comment by pmaier1** [comment]
Excellent!

File: src/Phoenix.vue [hunk 1]
```diff
@@ -21,6 +21,23 @@
             <template v-if="sidebar.mainContentComponent" v-slot:mainContent>
               <component :is="sidebar.mainContentComponent" />
             </template>
+            <template v-if="isQuotaVisible" v-slot:footer>
+              <div class="uk-text-center">
+                <oc-progress
+                  :value="parseInt(quota.relative)"
```
Comment (LukasHirt) [inline src/Phoenix.vue:27]: Adjust in state

**Comment by LukasHirt** [comment]
- [ ] Discuss if quota should be visible from different extensions
- [ ] Test switching between extensions if the quota state is really available and how the load of it works

**Comment by LukasHirt** [comment]
@PVince81 I added the extension point to display content in the footer of the sidebar. The quota is now included via that extension point and is only displayed in files app.

File: apps/files/src/store/actions.js [hunk 2]
```diff
@@ -399,7 +400,20 @@ export default {
...
+            context.commit('CHECK_QUOTA', quota)
+
+            // Display quota in the sidebar
+            if (
+              !isPublicPage &&
+              !context.getters.currentFolder.isMounted() &&
+              quota.definition !== 'default' &&
+              quota.definition !== 'none'
+            ) {
+              context.commit('SET_SIDEBAR_FOOTER_CONTENT_COMPONENT', SidebarQuota, { root: true })
```
Comment (PVince81) [inline apps/files/src/store/actions.js:414]: is there a code path where maybe the quota might have changed and we need to set an empty Vue into that extension point ? I could imagine this happening if the user is switching between views/apps while the admin might have changed the quota from a true value to infinite. it's quite a corner case though

Comment (LukasHirt) [reply apps/files/src/store/actions.js:414]: Right - I added an `else` which will reset the component in there.

**Pull request merged**